lib/ansible/modules/web_infrastructure/jenkins_plugin.py

```diff
@@ -427,11 +427,13 @@ def install(self):
                 self.module.fail_json(
                     msg="Jenkins home directory doesn't exist.")
 
-            md5sum_old = None
-            if os.path.isfile(plugin_file):
-                # Make the checksum of the currently installed plugin
-                md5sum_old = hashlib.md5(
-                    open(plugin_file, 'rb').read()).hexdigest()
+            try:
+                sha1_old = hashlib.sha1(open(plugin_file, 'rb').read())
+            except Exception as err:
+                self.module.fail_json(
+                    msg="Cannot calculate SHA1 of the old plugin.",
+                    details="SHA1 calculation failed with: {0}".format(err)
+                )
```

Comment:
Why the additional changes beyond switching the hash algorithm?

This appears to change the behavior of the module. Before a non-existent `plugin_file` would result in the hash being `None`, but now it results in a module failure.

Reply:
Correct, this comes directly from the previous PR, (Se original msg). I have no problem removing it.